doc/specs/#190 - Proxy Support.md

```diff
@@ -0,0 +1,89 @@
+---
+author: Flor Chacon @florelis
+created on: 2024-02-07
+last updated: 2024-02-07
+issue id: 190
+---
+
+# Proxy support
+
+For [#190](https://github.com/microsoft/winget-cli/issues/190)
+
+## Abstract
+
+This spec describes a feature to specify a proxy for winget to use when connecting to the internet.
+
+## Solution Design
+
+A new command line argument will be added to specify a proxy to use during a particular invocation of winget.
+This functionality will first need to be enabled through an admin setting, similar to local manifests or hash override.
+
+An option to set a default proxy to use on every flow will be added, but it will require administrator permissions to be set.
+
+New Group Policy will also be added for IT admins to control the use of proxies.
+The policies will be similar to those we already have for sources, so that a specific proxy can be required or only a predefined set of proxies can be allowed.
+
+Proxies will not be used for the configuration features for now.
+
+If a proxy is configured, it will be used for accessing sources and downloading installers.
+Note that it will not affect the behavior of installers themselves, so an installation may still generate traffic outside of the proxy.
+
+## UI/UX Design
+
+We will add a command line argument taking the URI to the proxy.
+A separate parameter will be available to disable the use of proxy if there is a default set.
+
+```
+> winget settings --enable ProxyCommandLineArgument
+> winget install Contoso.App --proxy https://127.0.0.1:2345
+> winget install Contoso.App --no-proxy
+```
+
+To configure the default proxy, a new `proxy` subcommand will be added to the `settings` command, with options to `set` and `reset` the default.
+
+```
+> winget settings proxy set https://127.0.0.1:2345
+> winget settings proxy reset
+```
+
+The current default proxy will be added to the output `winget --info`.
+
+## Capabilities
+
+### Accessibility
+
+This should have no direct impact on accessibility.
+
+### Security
+
+There is a possibility of an attacker using a malicious proxy to tamper with the data received from the source, or with the contents of the installer file.
+This is not much different from the risks of using a public network.
+The following mitigating factors will be in place:
+* (New) The ability to set a default proxy will be restricted to administrators, to prevent attackers from adding a proxy without the user realizing.
+* (New) A Group Policy will be available to block the use of proxies, or limit it to an approved list.
```

Comment:
Should there also be a mention of the Group Policy to force a default proxy? I see it mentioned above, but not specifically included here
> New Group Policy will also be added for IT admins to control the use of proxies.
The policies will be similar to those we already have for sources, so that a specific proxy can be required or only a predefined set of proxies can be allowed.